packages/core/src/routes/sign-in-experience/custom-ui-assets/index.ts

```diff
@@ -0,0 +1,100 @@
+import { readFile } from 'node:fs/promises';
+
+import { uploadFileGuard, maxUploadFileSize } from '@logto/schemas';
+import { generateStandardId } from '@logto/shared';
+import { object, z } from 'zod';
+
+import { EnvSet } from '#src/env-set/index.js';
+import RequestError from '#src/errors/RequestError/index.js';
+import koaGuard from '#src/middleware/koa-guard.js';
+import SystemContext from '#src/tenants/SystemContext.js';
+import assertThat from '#src/utils/assert-that.js';
+import { getConsoleLogFromContext } from '#src/utils/console.js';
+import { buildAzureStorage } from '#src/utils/storage/azure-storage.js';
+import { getTenantId } from '#src/utils/tenant.js';
+
+import { type ManagementApiRouter, type RouterInitArgs } from '../../types.js';
+
+export default function customUiAssetsRoutes<T extends ManagementApiRouter>(
+  ...[router]: RouterInitArgs<T>
+) {
+  // TODO: Remove
+  if (!EnvSet.values.isDevFeaturesEnabled) {
+    return;
+  }
+
+  router.post(
+    '/sign-in-exp/default/custom-ui-assets',
+    koaGuard({
+      files: object({
+        file: uploadFileGuard.array().min(1),
+      }),
```

Comment:
can we upload multiple times? I was assume that we can only upload a single zip file

Reply:
You can. And you will get a new `customUiAssetId` every time you do it